src/components/UserSettings/GlobalSettings.js

```diff
@@ -148,147 +148,147 @@ class GlobalSettings extends Component {
     this.saveOptions(saveFields, id)
   }
 
-  buildSections (onChange, translatedLabels) {
+  buildSections (onChange) {
     const { draftValues } = this.state
     const { config, msg } = this.props
     const idPrefix = 'global-user-settings'
     return {
       [GENERAL_SECTION]: {
         title: msg.general(),
         fields: [
-          {
+          ((name) => ({
             title: msg.username(),
-            name: 'username',
-            body: <span>{config.userName}</span>,
-          },
-          {
+            name,
+            body: <span>{config[name]}</span>,
+          }))('userName'),
```

Comment:
IDK - this style makes it a bit more difficult to read at the expense of have fewer specific values.  Feels a bit over complicated.  Any strong opinions?

Reply:
I agree, that for this field it's overkill (forced by the framework). The change was required for editable fields. There we need to use the same identifier in 4 places: 
1. name(used for React `key` property and for confirmation/error dialog), 
2. id, 
3. `draft[name]` (reading value from the state) and 
4. `onChange(name)` (storing value in the state).

Ensuring that the same string is used in all those places has become a problem.
We could move/hide this in the `Settings*` layer however every widget consumes the input differently which adds extra complexity.

Reply:
So I'm meh about the whole thing.  I'm ok with it if you're ok with it.